### PR TITLE
Remove Optional parameters from model constructors

### PR DESCRIPTION
**Pull request by ShivangiReja**
This PR includes the following updates:
- Removed the optional parameter from the constructors of `VectorStoreCreationHelper`, `AssistantChatMessage`, `ChatFunction`, and `FunctionChatMessage`.
- Removed the optional `purpose` parameter from `FileClient.GetFilesAsync` and `FileClient.GetFiles` methods, and added overloads with `purpose` as a required parameter.

For the reference, here's the list where we had optional parameters: https://gist.github.com/ShivangiReja/029d9eb032ac60103827ea55f97ec4f7

### CODE REVIEW

File: .dotnet/src/Custom/Chat/FunctionChatMessage.cs [hunk 1]
```diff
@@ -20,12 +20,8 @@ public partial class FunctionChatMessage : ChatMessage
     /// <param name="functionName">
     ///     The name of the called function that this message provides information from.
     /// </param>
-    /// <param name="content">
-    ///     The textual content that represents the output or result from the called function. There is no format
-    ///     restriction (e.g. JSON) imposed on this content.
-    /// </param>
-    public FunctionChatMessage(string functionName, string content = null)
-        : base(ChatMessageRole.Function, content)
+    public FunctionChatMessage(string functionName)
```
Comment (joseharriaga) [inline .dotnet/src/Custom/Chat/FunctionChatMessage.cs:23]: I just noticed that `content` is actually a required parameter here (although it is nullable and that's probably the reason why it is confusing). Could you restore the `content` parameter but removed its default value of `null`? i.e.:

```csharp
public FunctionChatMessage(string functionName, string content)
```

And then, you can also restore the `CreateFunctionMessage` factory method in the Custom/Chat/ChatMessage.cs file. That one was correct in that it was not defaulting the `content` parameter to `null`.

File: .dotnet/src/Custom/Chat/ChatMessage.cs [hunk 2]
```diff
@@ -78,7 +78,7 @@ internal ChatMessage(ChatMessageRole role, IEnumerable<ChatMessageContentPart> c
         }
     }
 
-    internal ChatMessage(ChatMessageRole role, string content)
+    internal ChatMessage(ChatMessageRole role, string content = null)
```
Comment (joseharriaga) [inline .dotnet/src/Custom/Chat/ChatMessage.cs:81]: nit: The `content` is required in all cases except in `AssistantChatMessage` when there are either `ChatToolCalls` or a `FuctionCall`. Since this is the exception and not the rule, could you keep this `content` parameter as not defaulted and then explicitly pass `null` in those two cases? i.e., in the Custom/Chat/AssistantChatMessage.cs file:

```csharp
public AssistantChatMessage(IEnumerable<ChatToolCall> toolCalls)
    : base(ChatMessageRole.Assistant, null)
```

And also:

```csharp
public AssistantChatMessage(ChatFunctionCall functionCall)
    : base(ChatMessageRole.Assistant, null)
```

File: .dotnet/tests/Chat/ChatToolTests.cs [hunk 3]
```diff
@@ -234,7 +232,9 @@ public async Task FunctionsWork()
         Assert.That(argumentsJson.ContainsKey("month_name"));
         Assert.That(argumentsJson["month_name"].ToString().ToLowerInvariant(), Is.EqualTo("february"));
         messages.Add(new AssistantChatMessage(result.Value));
-        messages.Add(new FunctionChatMessage(GetFavoriteColorForMonthFunctionName, "chartreuse"));
+        var functionChatMessage = new FunctionChatMessage(GetFavoriteColorForMonthFunctionName);
```
Comment (joseharriaga) [inline .dotnet/tests/Chat/ChatToolTests.cs:235]: If you make the change in my other comment, you can undo this change.

File: .dotnet.azure/sdk/openai/Azure.AI.OpenAI/tests/ChatTests.Functions.cs [hunk 4]
```diff
@@ -123,11 +122,13 @@ public async Task SimpleFunctionCallWorks(FunctionCallTestType functionCallType)
 
         // Complete the function call
         messages.Add(new AssistantChatMessage(completion.FunctionCall));
-        messages.Add(new FunctionChatMessage(FUNCTION_TEMPERATURE.FunctionName, JsonSerializer.Serialize(new
+        var functionChatMessage = new FunctionChatMessage(FUNCTION_TEMPERATURE.FunctionName);
+        functionChatMessage.Content.Add(ChatMessageContentPart.CreateTextPart(JsonSerializer.Serialize(new
```
Comment (joseharriaga) [inline .dotnet.azure/sdk/openai/Azure.AI.OpenAI/tests/ChatTests.Functions.cs:126]: If you make the change in my other comment, you can undo this change.

File: .dotnet.azure/sdk/openai/Azure.AI.OpenAI/tests/ChatTests.Functions.cs [hunk 5]
```diff
@@ -245,12 +246,14 @@ public async Task SimpleFunctionCallWorksStreaming(FunctionCallTestType function
             //             us manual control are internal. So let's use JSON.
             var converted = ModelReaderWriter.Read<ChatFunctionCall>(BinaryData.FromString(JsonSerializer.Serialize(new { name = functionName, arguments = functionArgs.ToString() })));
             messages.Add(new AssistantChatMessage(converted));
-            messages.Add(new FunctionChatMessage(FUNCTION_TEMPERATURE.FunctionName, JsonSerializer.Serialize(new
+            var functionChatMessage = new FunctionChatMessage(FUNCTION_TEMPERATURE.FunctionName);
```
Comment (joseharriaga) [inline .dotnet.azure/sdk/openai/Azure.AI.OpenAI/tests/ChatTests.Functions.cs:249]: If you make the change in my other comment, you can undo this change.

File: .dotnet/src/Custom/Chat/ChatMessage.cs (hunk 2 above)
Comment (ShivangiReja) [reply .dotnet/src/Custom/Chat/ChatMessage.cs:81]: >could you keep this content parameter as not defaulted and then explicitly pass null in those two cases? i.e., in the Custom/Chat/AssistantChatMessage.cs file:

If we don't default to null here, then how can I call `base(ChatMessageRole.Assistant, null)` from AssistantChatMessage ctor? do you want me to cast it?

```csharp
public AssistantChatMessage(IEnumerable<ChatToolCall> toolCalls)
     : base(ChatMessageRole.Assistant, (string)null)
```

Comment (ShivangiReja) [reply .dotnet/src/Custom/Chat/ChatMessage.cs:81]:
```csharp
internal ChatMessage(ChatMessageRole role, string content = null)
{
...
}
```
This is an internal constructor, so we can set the default value of `content` to `null` and call it in every case, regardless of whether `content` is required or not. This way, we avoid the need for casting.

File: .dotnet/src/Custom/Chat/FunctionChatMessage.cs (hunk 1 above)
Comment (ShivangiReja) [reply .dotnet/src/Custom/Chat/FunctionChatMessage.cs]: Ah..the default value was null here, so thought it's an optional parameter.

**Review by joseharriaga — APPROVED** [review]
